### ISSUE TITLE
Wrong "D105: Missing docstring in magic method" for overload function

### ISSUE THREAD

**Issue by ITProKyle** [PyCQA/pydocstyle#525]
Similar to #419 but for class magic methods.

The following example results in the the two overloaded `__getitem__` methods to be flagged with D105. The expected result is that they are not flagged.

```python
from typing import Iterable, List, MutableSequence, TypeVar, overload

_T = TypeVar("_T")

class MyClass(MutableSequence(_T)):
    """Example class."""

    def __init__(self, iterable: Iterable[_T]) -> None:
        """Instantiate class."""
        self._data = list(iterable)

    @overload
    def __getitem__(self, __index: int) -> _T:
        ...

    @overload
    def __getitem__(self, __index: slice) -> List[_T]:
        ...

    def __getitem__(self, __index):
        """Get item by index."""
        return self._data[__index]
```

**Comment by jenstroeger** [comment]
Probably also related is issue https://github.com/PyCQA/pydocstyle/issues/635
